[GStreamer] Implement gstStructureGetString()

philn opened the pull request:
#### 6ca202068171d6cced2d5a381eba85074fdabbdb
<pre>
[GStreamer] Implement gstStructureGetString()
<a href="https://bugs.webkit.org/show_bug.cgi?id=277000">https://bugs.webkit.org/show_bug.cgi?id=277000</a>

Reviewed by Xabier Rodriguez-Calvar.

Usage of gst_structure_get_string() is no longer recommended. Use this new helper instead.
This patch includes drive-by changes, reducing usage of raw pointers and avoiding memory allocations
when possible (using StringView instead of GUniquePtr&lt;char&gt; in the GStreamerRtpTransceiverBackend).

* Source/WebCore/Modules/mediastream/gstreamer/GStreamerMediaEndpoint.cpp:
(WebCore::GStreamerMediaEndpoint::requestPad):
* Source/WebCore/Modules/mediastream/gstreamer/GStreamerRtpTransceiverBackend.cpp:
(WebCore::toRtpCodecCapability):
(WebCore::getMsidFromCurrentCodecPreferences):
(WebCore::GStreamerRtpTransceiverBackend::setCodecPreferences):
(): Deleted.
* Source/WebCore/Modules/mediastream/gstreamer/GStreamerStatsCollector.cpp:
(WebCore::RTCStatsReport::Stats::Stats):
(WebCore::RTCStatsReport::RtpStreamStats::RtpStreamStats):
(WebCore::RTCStatsReport::CodecStats::CodecStats):
(WebCore::RTCStatsReport::RemoteInboundRtpStreamStats::RemoteInboundRtpStreamStats):
(WebCore::RTCStatsReport::RemoteOutboundRtpStreamStats::RemoteOutboundRtpStreamStats):
(WebCore::RTCStatsReport::OutboundRtpStreamStats::OutboundRtpStreamStats):
(WebCore::RTCStatsReport::TransportStats::TransportStats):
(WebCore::iceCandidateType):
(WebCore::RTCStatsReport::IceCandidateStats::IceCandidateStats):
(WebCore::RTCStatsReport::IceCandidatePairStats::IceCandidatePairStats):
* Source/WebCore/Modules/mediastream/gstreamer/GStreamerWebRTCUtils.cpp:
(WebCore::toRTCEncodingParameters):
(WebCore::toRTCRtpSendParameters):
(WebCore::payloadTypeForEncodingName):
(WebCore::capsFromRtpCapabilities):
(WebCore::capsFromSDPMedia):
* Source/WebCore/Modules/mediastream/gstreamer/GStreamerWebRTCUtils.h:
* Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp:
(WebCore::gstStructureGet):
* Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.h:
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
(WebCore::MediaPlayerPrivateGStreamer::handleMessage):
(WebCore::MediaPlayerPrivateGStreamer::loadNextLocation):
* Source/WebCore/platform/graphics/gstreamer/eme/WebKitCommonEncryptionDecryptorGStreamer.cpp:
(transformCaps):
(transformInPlace):
* Source/WebCore/platform/graphics/gstreamer/mse/GStreamerMediaDescription.cpp:
(WebCore::GStreamerMediaDescription::extractCodecName const):
* Source/WebCore/platform/mediastream/gstreamer/GStreamerCaptureDeviceManager.cpp:
(WebCore::GStreamerCaptureDeviceManager::addDevice):
* Source/WebCore/platform/mediastream/gstreamer/GStreamerIncomingTrackProcessor.cpp:
(WebCore::GStreamerIncomingTrackProcessor::configure):
(WebCore::GStreamerIncomingTrackProcessor::incomingTrackProcessor):
* Source/WebCore/platform/mediastream/gstreamer/GStreamerVideoCapturer.cpp:
(WebCore::GStreamerVideoCapturer::reconfigure):
* Source/WebCore/platform/mediastream/gstreamer/RealtimeOutgoingAudioSourceGStreamer.cpp:
(WebCore::RealtimeOutgoingAudioSourceGStreamer::setPayloadType):
* Source/WebCore/platform/mediastream/gstreamer/RealtimeOutgoingVideoSourceGStreamer.cpp:
(WebCore::RealtimeOutgoingVideoSourceGStreamer::setPayloadType):
* Tools/Scripts/webkitpy/style/checkers/cpp.py:
(check_language):
* Tools/Scripts/webkitpy/style/checkers/cpp_unittest.py:
(CppStyleTest):
* Tools/TestWebKitAPI/Tests/WebCore/gstreamer/GStreamerTest.cpp:
(TestWebKitAPI::TEST_F):

Canonical link: <a href="https://commits.webkit.org/281559@main">https://commits.webkit.org/281559@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/918e62fcb17e4aa4ff57dc8e53a63fcf85a15d45

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/60334 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/39686 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/12893 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/64254 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/10866 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/62464 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/47358 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/11099 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/48849 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/7567 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/62365 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/36990 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/52270 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/29696 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/59860 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/33688 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/9497 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/9783 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/55573 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/9784 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/65986 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/4268 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/9622 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/56210 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/4286 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/52244 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/56378 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13390 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/3554 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/35496 "Built successfully") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/36577 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/37667 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/36321 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

philn commented on Source/WebCore/platform/mediastream/gstreamer/GStreamerCaptureDeviceManager.cpp:
```diff
@@ -187,9 +187,12 @@ const Vector<CaptureDevice>& GStreamerCaptureDeviceManager::captureDevices()
-    const char* klass = gst_structure_get_string(properties.get(), "device.class");
+    auto deviceClassValue = gstStructureGet<std::span<const char>>(properties.get(), "device.class"_s);
+    if (!deviceClassValue)
+        return;
```
this early return is triggerring test failures i think.

calvaris commented on Source/WebCore/Modules/mediastream/gstreamer/GStreamerRtpTransceiverBackend.cpp:
```diff
@@ -131,40 +131,40 @@ static inline WARN_UNUSED_RETURN ExceptionOr<GstCaps*> toRtpCodecCapability(cons
         auto* s = gst_caps_get_structure(currentCaps.get(), 0);
-        msid = GUniquePtr<char>(g_strdup(gst_structure_get_string(s, "a-msid")));
+        if (auto msIdValue = gstStructureGet<ASCIILiteral>(s, "a-msid"_s))
+            return *msIdValue;
```
I don't know if this case is correct because I don't know the life-time of these caps. If the structure disappears, the memory inside the ASCIILiteral disappears, if I am not mistaken, because nothing is copied. An ASCIILiteral is something you can safely pass through everything because the life time of the const char* literal is guaranteed by the compiler.

I guess we want to spare a string copy if we can and for that I would return maybe a `StringView` that gives you more the impression of needing to take care of the life time.

For cases like this I would directly ensure the string copy with a `<String>` and then you don't care about the life-time because the string is copied. And if you don't want to duplicate API for owned and non-owned strings, you can always call `StringView::toString` which is going to do the job for you.

philn replied:
`StringView` makes no difference in this context because when built from `ASCIILiteral` it just assigns its `const void* m_characters` from the span data. And it's more complex to use because.

> I don't know if this case is correct because I don't know the life-time of these caps. 

The caps belong to the transceiver for which we keep a ref in `GStreamerRtpTransceiverBackend`.

calvaris commented on Source/WebCore/Modules/mediastream/gstreamer/GStreamerStatsCollector.cpp:
```diff
@@ -40,23 +40,27 @@ namespace WebCore {
         timestamp = *value;
+
+    if (auto value = gstStructureGet<ASCIILiteral>(structure, "id"_s))
+        id = String(*value);
```
And here you have another dangeous/wrong instance. As `*value`  is an `ASCIILiteral`,  `String()` is going to invoke `StringImpl::create` and for a `ASCIILiteral` the implementation is `ALWAYS_INLINE static Ref<StringImpl> create(ASCIILiteral literal) { return createWithoutCopying(literal.span8()); }`. This is a big life-time problem.

And from what I see below, you have lots of this below.

calvaris commented on Source/WebCore/Modules/mediastream/gstreamer/GStreamerStatsCollector.cpp:
```diff
@@ -209,7 +212,7 @@ RTCStatsReport::TransportStats::TransportStats(const GstStructure* structure)
 }
 
-static inline RTCIceCandidateType iceCandidateType(const String& type)
+static inline RTCIceCandidateType iceCandidateType(StringView type)
```
Why not a `const StringView&`

calvaris commented on Source/WebCore/Modules/mediastream/gstreamer/GStreamerWebRTCUtils.cpp:
```diff
@@ -484,15 +486,15 @@ uint32_t UniqueSSRCGenerator::generateSSRC()
 }
 
-std::optional<int> payloadTypeForEncodingName(const char* encodingName)
+std::optional<int> payloadTypeForEncodingName(StringView encodingName)
```
const&

calvaris commented on Source/WebCore/Modules/mediastream/gstreamer/GStreamerWebRTCUtils.h:
```diff
@@ -281,7 +281,7 @@ class UniqueSSRCGenerator : public ThreadSafeRefCounted<UniqueSSRCGenerator> {
 };
 
-std::optional<int> payloadTypeForEncodingName(const char* encodingName);
+std::optional<int> payloadTypeForEncodingName(StringView encodingName);
```
const&

calvaris commented on Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.h:
```diff
@@ -275,6 +275,9 @@ GstElement* makeGStreamerBin(const char* description, bool ghostUnlinkedPads);
 template<typename T>
 std::optional<T> gstStructureGet(const GstStructure*, ASCIILiteral key);
 
+template<typename T>
+std::optional<T> gstStructureGet(const GstStructure*, StringView key);
```
Stylistic nit: I would remove the empty line between the two declarations.

calvaris commented on Source/WebCore/platform/graphics/gstreamer/eme/WebKitCommonEncryptionDecryptorGStreamer.cpp:
```diff
@@ -153,8 +153,10 @@ static GstCaps* transformCaps(GstBaseTransform* base, GstPadDirection direction,
-                gst_structure_set_name(outgoingStructure.get(), gst_structure_get_string(outgoingStructure.get(), "original-media-type"));
+            if (!canDoPassthrough) {
+                if (auto originalMediaType = WebCore::gstStructureGet<ASCIILiteral>(outgoingStructure.get(), "original-media-type"_s))
+                    gst_structure_set_name(outgoingStructure.get(), originalMediaType->characters());
```
I think it would be better to not `if` here and just `ASSERT` (or even `RELEASE_ASSERT`) on the result for extra safety.

calvaris commented on Source/WebCore/platform/mediastream/gstreamer/GStreamerVideoCapturer.cpp:
```diff
@@ -279,7 +279,7 @@ void GStreamerVideoCapturer::reconfigure()
     struct MimeTypeSelector {
         const char* mimeType = "video/x-raw";
-        const char* format = nullptr;
+        std::optional<ASCIILiteral> format;
```
I think this should become a `<String>`

philn replied:
Why?

This made me realize we call `gst_caps_set_simple()` with an `optional<ASCIILiteral>` that should be fixed :)

calvaris replied:
It should become a `<String>` because you're keeping the format in the struct and you don't want to mess with the life time of the information you have gotten from the structure and also, conceptually, it is not a literal.

calvaris reviewed (changes requested):
Required changes can be trivial but given the complexity of the patch, I think it would require another review pass, hence the r- .

calvaris commented on Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp:
```diff
@@ -1040,13 +1040,61 @@ std::optional<T> gstStructureGet(const GstStructure* structure, ASCIILiteral key
 }
 
+template<typename T>
+std::optional<T> gstStructureGet(const GstStructure* structure, StringView key)
```
As I said before, you can convert make the implementation go the one taking ASCIILiteral as a parameter and base that implementation in this one. Having two parallel implementations just because of a so similar parameter is not good. Return value should be always StringView.

I find this very important.

calvaris commented on Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp:
```diff
@@ -1040,13 +1040,61 @@ std::optional<T> gstStructureGet(const GstStructure* structure, ASCIILiteral key
+StringView gstStructureGetString(const GstStructure* structure, StringView key)
+{
+    return StringView::fromLatin1(gst_structure_get_string(structure, key.toStringWithoutCopying().ascii().data()));
+}
```
Why did you create new functions instead of just using gstStructureGet<StringView> ? I think it is a good idea to keep the same structure as in the other cases. I guess your answer could be that with the isNull check (you implicitly use when invoking the bool operator) is enough and I guess I can be satisfied with the answer.

philn replied:
Yes, `std::optional` wrapping a `StringView` is redundant.

calvaris replied:
It depends on if you want to distinguish between not having a property or considering null properties. But I guess in this case it does not matter and we can go as it is.

philn replied:
afaik a GstStructure can't have null string fields. The string is either empty or not.

calvaris commented on Source/WebCore/Modules/mediastream/gstreamer/GStreamerRtpTransceiverBackend.cpp:
```diff
@@ -109,7 +109,7 @@ bool GStreamerRtpTransceiverBackend::stopped() const
 }
 
-static inline WARN_UNUSED_RETURN ExceptionOr<GstCaps*> toRtpCodecCapability(const RTCRtpCodecCapability& codec, int& dynamicPayloadType, const char* msid)
+static inline WARN_UNUSED_RETURN ExceptionOr<GRefPtr<GstCaps>> toRtpCodecCapability(const RTCRtpCodecCapability& codec, int& dynamicPayloadType, StringView msid)
```
Please mention this drive-by change in the changelog.

calvaris commented on Source/WebCore/Modules/mediastream/gstreamer/GStreamerStatsCollector.cpp:
```diff
@@ -40,23 +40,27 @@ namespace WebCore {
         timestamp = *value;
+
+    if (auto value = gstStructureGetString(structure, "id"_s))
+        id = makeString(value);
```
you can also do toString() from the StringView.

calvaris commented on Source/WebCore/Modules/mediastream/gstreamer/GStreamerWebRTCUtils.cpp:
```diff
@@ -592,8 +595,8 @@ GRefPtr<GstCaps> capsFromSDPMedia(const GstSDPMedia* media)
-            if (const char* name = gst_structure_get_string(structure, "encoding-name")) {
-                auto encodingName = String(span(name)).convertToASCIIUppercase();
+            if (auto name = gstStructureGetString(structure, "encoding-name"_s)) {
+                auto encodingName = makeString(name).convertToASCIIUppercase();
```
You don't need to convert it to String first, there's StringView::convertToASCIIUppercase .

calvaris commented on Source/WebCore/platform/graphics/gstreamer/eme/WebKitCommonEncryptionDecryptorGStreamer.cpp:
```diff
@@ -153,8 +153,11 @@ static GstCaps* transformCaps(GstBaseTransform* base, GstPadDirection direction,
+            if (!canDoPassthrough) {
+                auto originalMediaType = WebCore::gstStructureGetString(outgoingStructure.get(), "original-media-type"_s);
+                RELEASE_ASSERT(originalMediaType);
+                gst_structure_set_name(outgoingStructure.get(), originalMediaType.toStringWithoutCopying().ascii().data());
```
I think you don't need to convert to String to extract the characters

calvaris commented on Source/WebCore/platform/mediastream/gstreamer/GStreamerIncomingTrackProcessor.cpp:
```diff
@@ -66,10 +66,11 @@ void GStreamerIncomingTrackProcessor::configure(ThreadSafeWeakPtr<GStreamerMedia
+        if (auto msIdAttribute = gstStructureGetString(structure, msIdAttributeName)) {
+            auto components = makeString(msIdAttribute).split(' ');
+            if (components.size() == 2)
+                m_sdpMsIdAndTrackId = { components[0], components[1] };
```
You can use split with a StringView even when dealing with the result can be more complex. If you don't to deal with that complexity, you can still spare one copy if you create the string with toStringWithoutCopying .

philn replied:
I find this version simpler. Dealing with a `SplitResult` here is cumbersome.

calvaris replied:
Ok, but then create the String without copying.

calvaris commented on Source/WebCore/platform/mediastream/gstreamer/GStreamerIncomingTrackProcessor.cpp:
```diff
@@ -175,7 +176,7 @@ GRefPtr<GstElement> GStreamerIncomingTrackProcessor::incomingTrackProcessor()
         auto structure = gst_caps_get_structure(m_data.caps.get(), 0);
         ASSERT(gst_structure_has_name(structure, "application/x-rtp"));
-        String encodingNameValue = WTF::span(gst_structure_get_string(structure, "encoding-name"));
+        auto encodingNameValue = makeString(gstStructureGetString(structure, "encoding-name"_s));
```
You can convert from the StringView, as mentioned above.

calvaris commented on Source/WebCore/platform/mediastream/gstreamer/RealtimeOutgoingAudioSourceGStreamer.cpp:
```diff
@@ -61,8 +61,10 @@ bool RealtimeOutgoingAudioSourceGStreamer::setPayloadType(const GRefPtr<GstCaps>
-    if (encoding.isNull()) {
+    String encoding;
+    if (auto encodingName = gstStructureGetString(structure.get(), "encoding-name"_s))
+        encoding = encodingName.convertToASCIILowercase();
```
see? you already knew you can convert!

calvaris commented on Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp:
```diff
@@ -1047,6 +1054,23 @@ template std::optional<uint64_t> gstStructureGet(const GstStructure*, ASCIILiter
+
+StringView gstStructureGetString(const GstStructure* structure, StringView key)
+{
+    return StringView::fromLatin1(gst_structure_get_string(structure, key.toStringWithoutCopying().ascii().data()));
```
You can probably get the characters without converting to string first.

philn committed:
[GStreamer] Implement gstStructureGetString()
https://bugs.webkit.org/show_bug.cgi?id=277000

Reviewed by Xabier Rodriguez-Calvar.

Usage of gst_structure_get_string() is no longer recommended. Use this new helper instead.
This patch includes drive-by changes, reducing usage of raw pointers and avoiding memory allocations
when possible (using StringView instead of GUniquePtr<char> in the GStreamerRtpTransceiverBackend).

* Source/WebCore/Modules/mediastream/gstreamer/GStreamerMediaEndpoint.cpp:
(WebCore::GStreamerMediaEndpoint::requestPad):
* Source/WebCore/Modules/mediastream/gstreamer/GStreamerRtpTransceiverBackend.cpp:
(WebCore::toRtpCodecCapability):
(WebCore::getMsidFromCurrentCodecPreferences):
(WebCore::GStreamerRtpTransceiverBackend::setCodecPreferences):
(): Deleted.
* Source/WebCore/Modules/mediastream/gstreamer/GStreamerStatsCollector.cpp:
(WebCore::RTCStatsReport::Stats::Stats):
(WebCore::RTCStatsReport::RtpStreamStats::RtpStreamStats):
(WebCore::RTCStatsReport::CodecStats::CodecStats):
(WebCore::RTCStatsReport::RemoteInboundRtpStreamStats::RemoteInboundRtpStreamStats):
(WebCore::RTCStatsReport::RemoteOutboundRtpStreamStats::RemoteOutboundRtpStreamStats):
(WebCore::RTCStatsReport::OutboundRtpStreamStats::OutboundRtpStreamStats):
(WebCore::RTCStatsReport::TransportStats::TransportStats):
(WebCore::iceCandidateType):
(WebCore::RTCStatsReport::IceCandidateStats::IceCandidateStats):
(WebCore::RTCStatsReport::IceCandidatePairStats::IceCandidatePairStats):
* Source/WebCore/Modules/mediastream/gstreamer/GStreamerWebRTCUtils.cpp:
(WebCore::toRTCEncodingParameters):
(WebCore::toRTCRtpSendParameters):
(WebCore::payloadTypeForEncodingName):
(WebCore::capsFromRtpCapabilities):
(WebCore::capsFromSDPMedia):
* Source/WebCore/Modules/mediastream/gstreamer/GStreamerWebRTCUtils.h:
* Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp:
(WebCore::gstStructureGet):
* Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.h:
* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
(WebCore::MediaPlayerPrivateGStreamer::handleMessage):
(WebCore::MediaPlayerPrivateGStreamer::loadNextLocation):
* Source/WebCore/platform/graphics/gstreamer/eme/WebKitCommonEncryptionDecryptorGStreamer.cpp:
(transformCaps):
(transformInPlace):
* Source/WebCore/platform/graphics/gstreamer/mse/GStreamerMediaDescription.cpp:
(WebCore::GStreamerMediaDescription::extractCodecName const):
* Source/WebCore/platform/mediastream/gstreamer/GStreamerCaptureDeviceManager.cpp:
(WebCore::GStreamerCaptureDeviceManager::addDevice):
* Source/WebCore/platform/mediastream/gstreamer/GStreamerIncomingTrackProcessor.cpp:
(WebCore::GStreamerIncomingTrackProcessor::configure):
(WebCore::GStreamerIncomingTrackProcessor::incomingTrackProcessor):
* Source/WebCore/platform/mediastream/gstreamer/GStreamerVideoCapturer.cpp:
(WebCore::GStreamerVideoCapturer::reconfigure):
* Source/WebCore/platform/mediastream/gstreamer/RealtimeOutgoingAudioSourceGStreamer.cpp:
(WebCore::RealtimeOutgoingAudioSourceGStreamer::setPayloadType):
* Source/WebCore/platform/mediastream/gstreamer/RealtimeOutgoingVideoSourceGStreamer.cpp:
(WebCore::RealtimeOutgoingVideoSourceGStreamer::setPayloadType):
* Tools/Scripts/webkitpy/style/checkers/cpp.py:
(check_language):
* Tools/Scripts/webkitpy/style/checkers/cpp_unittest.py:
(CppStyleTest):
* Tools/TestWebKitAPI/Tests/WebCore/gstreamer/GStreamerTest.cpp:
(TestWebKitAPI::TEST_F):

Canonical link: https://commits.webkit.org/281559@main
